mypy/newsemanal/semanal.py

```diff
@@ -2803,6 +2803,10 @@ def process_typevar_parameters(self, args: List[Expression],
     def basic_new_typeinfo(self, name: str, basetype_or_fallback: Instance) -> TypeInfo:
         class_def = ClassDef(name, Block([]))
         class_def.fullname = self.qualified_name(name)
+        if self.is_func_scope() and not self.type:
```

Comment:
This would be better as an if/else statement.

Reply:
OK.